### PR TITLE
DRIVERS-2882 Add scripts to test AKS

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/auth_oidc/azure/create-and-setup-vm.sh [hunk 1]
```diff
@@ -31,19 +31,6 @@ export AZUREKMS_TENANTID=$AZUREOIDC_TENANTID
 export AZUREKMS_SECRET=$AZUREOIDC_SECRET
 export AZUREKMS_CLIENTID=$AZUREOIDC_APPID
 
-# Check for Azure Command-Line Interface (`az`) version 2.25.0 or newer.
```
Comment (blink1073) [inline .evergreen/auth_oidc/azure/create-and-setup-vm.sh:34]: This block was moved to the login file.

File: .evergreen/auth_oidc/k8s/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## OIDC on K8S
+
+- Launch an Atlas cluster
```
Comment (blink1073) [inline .evergreen/auth_oidc/k8s/README.md:3]: This is a placeholder for now, I still need to add GKE and EKS

File: .evergreen/csfle/azurekms/create-and-setup-vm.sh [hunk 3]
```diff
@@ -47,19 +47,6 @@ done
 # Set defaults.
 export AZUREKMS_IMAGE=${AZUREKMS_IMAGE:-"Debian:debian-11:11:0.20221020.1174"}
 
-# Check for Azure Command-Line Interface (`az`) version 2.25.0 or newer.
```
Comment (blink1073) [inline .evergreen/csfle/azurekms/create-and-setup-vm.sh:50]: This moved to the login file.

File: .evergreen/csfle/azurekms/remote-scripts/setup-azure-vm.sh [hunk 4]
```diff
@@ -4,14 +4,12 @@ set -o pipefail
 # Do not error on unset variables. run-orchestration.sh accesses unset variables.
 
 echo "Installing dependencies ... begin"
-# Make apt-get non-interactive.
-echo "debconf debconf/frontend select noninteractive" | sudo debconf-set-selections
 # Skip the "Processing triggers for man-db" step.
 echo "set man-db/auto-update false" | sudo debconf-communicate; sudo dpkg-reconfigure -f noninteractive man-db
 sudo apt-get -qq update
-OPTIONS="-y -qq -o DPkg::Lock::Timeout=-1"
+OPTIONS="-qq -y -o DPkg::Lock::Timeout=-1"
```
Comment (blink1073) [inline .evergreen/csfle/azurekms/remote-scripts/setup-azure-vm.sh:10]: I cleaned up a bunch of these apt-get calls to actually be quiet

File: .evergreen/ensure-binary.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  exit 0
+fi
+
+OS_NAME=$(uname -s | tr '[:upper:]' '[:lower:]')
+MARCH=$(uname -m | tr '[:upper:]' '[:lower:]')
+TARGET=${DRIVERS_TOOLS}/.bin/$NAME
+URL=""
+
+case $NAME in
+  kubectl)
```
Comment (blink1073) [inline .evergreen/ensure-binary.sh:25]: I'll be adding other binaries to this list in further PRs: gcloud and eksctl

File: .evergreen/k8s/aks/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#
+
+## AKS Cluster Setup
```
Comment (blink1073) [inline .evergreen/k8s/aks/README.md:3]: Again this is a stub for now

File: .evergreen/k8s/setup-pod.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -eux
+
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../handle-paths.sh
+pushd $SCRIPT_DIR
```
Comment (NoahStapp) [inline .evergreen/k8s/setup-pod.sh:6]: No matching `popd` in this file.

File: .evergreen/auth_oidc/k8s/remote-scripts/run-self-test.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+echo "Installing dependencies ... begin"
+git clone https://github.com/mongodb/mongo-python-driver
+pushd mongo-python-driver
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -U -q pip
+pip install -U -q requests
+pip install .
+popd
+echo "Installing dependencies ... end"
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/k8s/remote-scripts/run-self-test.sh:13]: Actually we don't need requests for this one, removed that part.

File: .evergreen/auth_oidc/k8s/run-driver-test.sh [hunk 9]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o pipefail
+set -o nounset
+
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../../handle-paths.sh
+pushd $SCRIPT_DIR
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/k8s/run-driver-test.sh:8]: Fixed all three

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**